### ISSUE TITLE
ValueError: Attempted relative import in non-package

### ISSUE THREAD

**Issue by jrussell88** [pazaan/decoding-contour-next-link#25]
The command:

> /git/decoding-contour-next-link$ python -m read_minimed_next24

fails with the error:
> Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/media/john/Home/john/git/decoding-contour-next-link/read_minimed_next24.py", line 22, in <module>
    from .pump_history_parser import NGPHistoryEvent, BloodGlucoseReadingEvent
ValueError: Attempted relative import in non-package

Actually I wanted to run:
> python get_hmac_and_key.py

to check my meter's AES key, but that gets a similar error:
> Traceback (most recent call last):
  File "get_hmac_and_key.py", line 18, in <module>
    from read_minimed_next24 import Config, Medtronic600SeriesDriver
  File "/media/john/Home/john/git/decoding-contour-next-link/read_minimed_next24.py", line 22, in <module>
    from .pump_history_parser import NGPHistoryEvent, BloodGlucoseReadingEvent
ValueError: Attempted relative import in non-package

Can anyone suggest a fix?

**Comment by mortlind** [comment]
Hi @jrussell88, The whole repo is now usable as a library because package-local modules are imported relative to the top-level package; i.e. the library. This, in turn, invalidates direct use of those local module when run as ordinary scripts. These scripts need now be run as modules, like you do it. However, the top-level package must now be in the Python path. Thus either you install the library somewhere in your Python path, e.g. inside ~/.local/lib/python<major>.<minor>/site-packages/ on linux, or you place your console CWD on top of the library folder. 

The whole matter is about avoiding ambiguity. Imagine there was a `string.py` module in the library, then `import string` would be ambiguous from a sibling module in the library. The current resolution is to always use relative imports when modules import modules from the same library. See PEP328 (https://www.python.org/dev/peps/pep-0328/).

To sum it up: Have a console with CWD in the directory where `decoding-contour-next-link` is a sub-directory. Execute the module ` read_minimed_next24` as follows: 
```
$ python3 -m decoding-contour-next-link.read_minimed_next24
```

**Comment by jrussell88** [comment]
Thanks for the quick reply and clear explanation. 

I ran the command from the parent folder of /decoding-contour-next-link, which generated:

> ImportError: No module named 'hid'

As I assume these errors are unrelated to the Current Working Directory issue I'll open a fresh issue.

**Comment by mortlind** [comment]
@jrussell88 If you are now running with Python3 whereas you formerly was set up with Python2, you must install the samme requirements with `pip3` as you did with `pip`. The latter installs for Python2, since that is, probably, currently the default Python interpreter of your system.

**Comment by jrussell88** [comment]
Ah, I thought it must be a configuration problem. I did try running with Python - my default is 2.7 - but that also failed. 
Didn't realise this required Python3. Looks like I'm back to chasing missing dependencies. I'll close this

**Comment by mortlind** [comment]
It does not require Python3, it supposedly works for Python2 also. Try the following in the library directory if you wish to set up for Python3:
`# pip3 install -r requirements.txt`

**Comment by jrussell88** [comment]
I reinstalled the dependencies for pip3; 

> $ sudo -H pip3 install cython
$ sudo -H pip3 install hidapi
$ sudo -H pip3 install requests astm PyCrypto crc16 python-dateutil
$ sudo -H pip3 install python-lzo

Then:

> /git$ pip3 install -r requirements.txt

As I was still getting file errors, I ran the commands as root:

> /git$ sudo python -m decoding-contour-next-link.read_minimed_next24

which worked. Does this suggest a problem with my Python3 permissions? I haven't run across this issue before.

However:

> /git$ sudo python3 -m decoding-contour-next-link.get_hmac_and_key.p

continues to return an _invalid syntax_ error.

Thanks very much for all your help.

**Comment by mortlind** [comment]
@jrussell88 As previously stated, `get_hmac_and_key.py` is not necessary any longer and it is not ported to Python3, so just forget about it.

**Comment by jrussell88** [comment]
Thanks I don't think I saw that. 
Is there any way to capture the key?

**Comment by mortlind** [comment]
@jrussell88 I think the key is included, so no need to get it.
